File: numberfacts/src/main/java/com/zdvdev/numberfacts/widget/ShadowSpan.java
package com.zdvdev.numberfacts.widget;

import android.text.TextPaint;
import android.text.style.CharacterStyle;

public class ShadowSpan extends CharacterStyle {
	public float mDx;
	public float mDy;
	public float mRadius;
	public int mColor;

	public ShadowSpan(float radius, float dx, float dy, int color) {
		this.mRadius = radius;
		this.mDx = dx;
		this.mDy = dy;
		this.mColor = color;
	}

	@Override public void updateDrawState(TextPaint tp) {
		tp.setShadowLayer(mRadius, mDx, mDy, mColor);
	}
}

File: numberfacts/src/main/java/com/zdvdev/numberfacts/widget/ShadowSpan.kt
package com.zdvdev.numberfacts.widget

import android.text.TextPaint
import android.text.style.CharacterStyle

class ShadowSpan(var radius: Float, var dx: Float, var dy: Float, var color: Int) : CharacterStyle() {
    override fun updateDrawState(tp: TextPaint) = tp.setShadowLayer(radius, dx, dy, color)
}